### PR TITLE
Disply logs for failed Jobs.

### LINKED ISSUE #20: Print logs from failed jobs

### PR DESCRIPTION
**Pull request by Behoston**
resolves: https://github.com/socialwifi/kubepy/issues/20

This change is [Reviewable](https://reviewable.io/reviews/socialwifi/kubepy/32)

### CODE REVIEW

File: kubepy/appliers.py [hunk 1]
```diff
@@ -174,8 +177,26 @@ def raise_if_failed(self):
                 self._raise_for_failed_condition(condition)
         failures = self.status.get('failed', 0)
         if self.max_retries and failures > self.max_retries:
+            failed_pod = self.get_job_pod()
+            if failed_pod:
+                stdout, stderr = self.get_failed_pod_logs(failed_pod)
+            else:
+                stdout, stderr = '', ''
+            for line in str(stdout).split('\\n'):
```
Comment (jgoclawski) [inline kubepy/appliers.py]: It's a bit overcomplicated to work on raw `bytes` type. 
You can go with:

```
  for line in stdout.decode().splitlines():
```

I'd also drop prefixing every line with stdout. kubectl logs mixes stdout and stderr anyway ([28167](https://github.com/kubernetes/kubernetes/issues/28167)), so what you call `stderr` here is not caontainer's `stderr` but `kubectl logs`'s `stderr` :)
That would leave us with a clean:
```
  print(stdout.decode())
```

File: kubepy/appliers.py [hunk 2]
```diff
@@ -174,8 +177,26 @@ def raise_if_failed(self):
...
+                print('STDOUT:\t', line)
+            for line in str(stderr).split('\\n'):
+                print('STDERR:\t', line)
             raise JobError('Job failed {} times.'.format(failures))
 
+    def get_failed_pod_logs(self, failed_pod):
+        for init_container_status in failed_pod['status'].get('initContainerStatuses', []):
+            if init_container_status['state'].get('terminated', {}).get('reason') == 'ContainerCannotRun':
+                return 'Init container failed!', init_container_status['state']['terminated']['message']
```
Comment (jgoclawski) [inline kubepy/appliers.py]: It's misleading to return an arbitrary string and pretend it came from stdout.
As mentioned above: maybe let's drop stdout/stderr distinction?

**Review by jgoclawski — CHANGES_REQUESTED** [review]

**Comment by pguz** [comment]
Hi,

I have just made some tests using this pull and everything seems to work. I look forward to see these changes merged.

Have a nice day,
Paweł

**Review by jgoclawski — APPROVED** [review]
LGTM :+1:

**Pull request merged**